studio/components/interfaces/Settings/Vault/Keys/EncryptionKeysManagement.tsx

```diff
@@ -0,0 +1,327 @@
+import dayjs from 'dayjs'
+import { observer } from 'mobx-react-lite'
+import * as Tooltip from '@radix-ui/react-tooltip'
+import { FC, Fragment, useEffect, useState } from 'react'
+import {
+  Alert,
+  Input,
+  IconSearch,
+  Listbox,
+  Button,
+  Modal,
+  Form,
+  IconTrash,
+  Badge,
+  IconKey,
+  IconLoader,
+  IconX,
+} from 'ui'
+import { useStore } from 'hooks'
+import Divider from 'components/ui/Divider'
+import { useRouter } from 'next/router'
+
+const DEFAULT_KEY_NAME = 'No description provided'
+
+interface Props {}
+
+const EncryptionKeysManagement: FC<Props> = ({}) => {
+  const { vault, ui } = useStore()
+  const router = useRouter()
+  const { id } = router.query
```

Comment:
nit: `useParams()` here instead of these two lines and then no need to do `id as string` below :)